include/AutomatableModel.h

```diff
@@ -170,6 +170,23 @@ class LMMS_EXPORT AutomatableModel : public Model, public JournallingObject
 		return castValue<T>( m_value );
 	}
 
+	inline float getTrueValue(int frameOffset = 0) const
+	{
+		if (m_controllerConnection != nullptr)
+		{
+			if (m_useControllerValue == true)
+			{
+				return controllerValue(frameOffset);
+			}
+		}
+		else if (hasLinkedModels() == true)
+		{
+			return controllerValue(frameOffset);
+		}
+
+		return m_value;
```

Comment:
You can invert this check for better readability.

Reply:
If m_controllerConnection is null, return the value, then handle the rest

Reply:
Just copied the old code. I will fix that.